### PR TITLE
Readded device_assert skipping in index and index_put (and also added

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112093

copy to noop pass)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

Differential Revision: [D50821678](https://our.internmc.facebook.com/intern/diff/D50821678)

### CODE REVIEW

**Review by oulgen — APPROVED** [review]

File: torch/_inductor/lowering.py
Comment (lezcano) [inline torch/_inductor/lowering.py:2920]: Either change all the `check` parameters to `add_asserts` (a better name IMO) or name this one `check` as well.

**Review by lezcano — APPROVED** [review]
The device_assert change LGTM modulo a notation nit. The other changes seem unrelated so they may belong to a different PR?

**Comment by Chillee** [comment]
@lezcano I'm uhh... smuggling two changes in here. The other change is pretty minor though.

**Comment by Chillee** [comment]
@izaitsevfb lmk when I can reland.

**Comment by izaitsevfb** [comment]
> @izaitsevfb lmk when I can reland.

After we either revert #108690 or re-land it internally (it fails some tests, but at this point that might be a better option).

**Comment by malfet** [comment]
> @pytorchbot revert -m "Stack breaks internal builds" -c ghfirst

Looks like it breaks some fbgemm_gpu and torchbench test cases.

**Comment by Chillee** [comment]
@chillee has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D50821678).

**Comment by Chillee** [comment]
Landed in https://github.com/pytorch/pytorch/pull/112476

**Pull request closed without merging**
